### ISSUE TITLE
tag vs release

### ISSUE THREAD

**Issue by nmizukami** [ESCOMP/mizuRoute#253]
can you tag the commit but no release?
if so, what is the rule of thumb for only tag (no release) and both?

**Comment by ekluzek** [comment]
Yes, but I think the only way to do that is through the command line. So we typically tag with something like...

git tag -a v2.0.1
git push escomp v2.0.1

The "-a" means an annotated tag that has you add the description of what you are doing. This is recommended for tags. Then you push it to the main fork (so here I assume there is a remote setup for escomp to goto ESCOMP/mizuroute.git.

release is for more stable more highly tested configurations. For CTSM we only release public versions that we tag as "release-*". You can also tag a release as "pre-release" which we do for release tags that aren't in a CESM release version.

**Comment by ekluzek** [comment]
One thing worthy of note that I did see. The current tags seem to be "unannotated" or "lightweight" tags. Annotated tags are better because they also do a commit for them. So you see a hash for the specific commit in the commit log for the tag, and not just a pointer to the commit the tag points to.
